angular/src/app/landing/data-files/data-files.component.ts

```diff
@@ -401,21 +401,19 @@ export class DataFilesComponent implements OnInit, OnChanges {
      */
     treeTableToggled(event: any) {
         //Set tree table's height based on the tree status
-        // If any node was expanded, set the height to max height
-        //Otherwise if files.length = 1, set height to min
+        // If any node was expanded, and table height is minimum height, set it to maximum height.
+        //Otherwise leave the table height as it is.
         let expanded: boolean = false;
         this.files.forEach((file) => {
             if(file.expanded) expanded = true;
         })
         this.isExpanded = expanded;
 
-        if(this.isExpanded){
-            this.treeTableHeight = MaxTreeTableHeight;
+        if(this.files.length <= 1) {
```

Comment:
It looks like this causes the view height to always shrink to the minimum if there is only one top-level item, even if that item is a folder.  

I think what you want is something like this (with no `else` section):
```typescript
if (this.filesCount <= 1  || this.treeTableHeight < MinTreeTableHeight) 
    this.treeTableHeight = MinTreeTableHeight;
```

Reply:
In the case that there is only one top-level item, and that item is a folder, and the table height is MinTreeTableHeight, if user expand the top folder, do we want to expand the window accordingly? This is what the else section try to do.